src/commands/hug.js

```diff
@@ -1,8 +1,66 @@
+import axios from 'axios';
+import Discord from 'discord.js';
 import DiscordCommand from './command.js';
 
 export default class HugCommand extends DiscordCommand {
-    async execute() {
-        const hug = this.getEmojiByName('HugSpin');
-        this.message.channel.send(`${hug}`);
+
+  async execute() {
+    if (!this.member) {
+      this.send("You must be in a Roo Bot enabled Discord server to use this command");
+      return;
+    }
+
+    const hug = this.getEmojiByName('HugSpin');
+    this.message.react(hug);
+
+    let receivers = "";
+    const mentionedUsernames = [];
+
+    if (this.hasRoleMentions || this.hasChannelMentions) {
+      this.send(`Please use \`\`\`${DiscordCommand.prefix}hug @user @user...\`\`\``);
+      return;
     }
+
+    if (this.hasUserMentions) {
+      if (this.args.length == this.userMentions.size) {
+        this.userMentions.each(user => {
+          if (user.displayName == this.member.displayName) {
+            mentionedUsernames.push("themself");
+          } else {
+            mentionedUsernames.push(user.displayName);
+          }
+        });
+        receivers = [
+          mentionedUsernames.slice(0, -1).join(', '),
+          mentionedUsernames.slice(-1)[0]
+        ].join(mentionedUsernames.length < 2 ? '' : ' and ');
```

Comment:
this is honestly pretty mind-bendy haha

maybe something a bit more verbose, but clear

```
const needsAnd = usersToMention.length > 1;
const lastUsername = mentionedUsernames[mentionedUsernames.length - 1];
const commaDelimitedUsers = needsAnd ?  mentionedUsernames.slice(0, -1) : mentionedUsernames;
receiver = needsAnd ? `${commaDelimitedUsers.join(', ')} and {lastUserName}` : `${commaDelimitedUsers.join(', ')}`; 
```